lib/generateFactory.js

```diff
@@ -41,15 +41,16 @@ function generateFactory(namespaces) {
       cw.blComment(() => {
         cw.ln('Create an instance of a class from its FHIR representation.')
           .ln('@param {Object} fhir - The element data in FHIR format (use `{}` and provide `type` for a blank instance)')
-          .ln(`@param {string} [type] - The (optional) type of the element (e.g., 'http://standardhealthrecord.org/spec/shr/demographics/PersonOfRecord').  This is only used if the type cannot be extracted from the JSON.`)
+          .ln(`@param {string} fhirType - the type of the FHIR object that was passed in, in case not otherwise identifiable from the object itself`)
```

Comment:
Is `fhirType` optional?  I noticed you pass in `null` for it in some of the cases above.  So it seems it must be optional?  The `shrType` below explicitly states that it is optional (in the description).

Also note that in jsdoc, you can indicate a parameter is optional by surrounding it in `[` `]`.  See: https://jsdoc.app/tags-param.html#optional-parameters-and-default-values

Reply:
Actually, I see you did use the `[` `]` jsdoc syntax in the namespace factory... so I guess you knew.  Maybe you missed it here?

Reply:
Yeah I think I first learned I could do that between here and the namespace factory. For consistency I'll change the namespace factory comment to remove the brackets. Technically whether it's optional or not will depend on the individual class, since most classes won't use it but some will require it.